File: buildSrc/src/main/groovy/org/mockito/release/comparison/PomComparator.java
package org.mockito.release.comparison;

import groovy.lang.Closure;

import static org.mockito.release.util.ArgumentValidation.notNull;

class PomComparator {

    private Closure<String> left;
    private Closure<String> right;

    PomComparator setPair(Closure<String> left, Closure<String> right) {
        notNull(left, "pom content to compare", right, "pom content to compare");
        this.left = left;
        this.right = right;
        return this;
    }

    boolean areEqual() {
        String left = this.left.call();
        String right = this.right.call();

        notNull(left, "pom content to compare", right, "pom content to compare");

        return replaceVersion(left).equals(replaceVersion(right));
    }

    private String replaceVersion(String pom) {
        return pom.replaceFirst("<version>(.*)</version>", "<version>foobar</version>");
    }
}

File: buildSrc/src/main/kotlin/org/mockito/release/comparison/PomComparator.kt
package org.mockito.release.comparison

import groovy.lang.Closure

import kotlin.text.replaceFirst
import kotlin.text.toRegex

internal class PomComparator(val left: Closure<String>, val right: Closure<String>) {

    fun areEqual(): Boolean {
        val leftValue = left.call()
        val rightValue = right.call()

        fun replaceVersion(pom: String): String {
            return pom.replaceFirst("<version>(.*)</version>".toRegex(), "<version>foobar</version>")
        }

        return replaceVersion(leftValue) == replaceVersion(rightValue)
    }
}
